frontend/readhistory.lua

```diff
@@ -40,14 +41,17 @@ local function timeFirstOrdering(l, r)
 end
 
 function ReadHistory:_indexing(start)
+    assert(self ~= nil)
```

Comment:
I have no strong opinion, since they are "private" functions anyway. If you prefer debug assert here, I will change it.